firmware/thruna/DCM/Src/App/App_Regen.c

```diff
@@ -0,0 +1,73 @@
+#include "App_Regen.h"
+
+static bool wheel_speed_in_range(void);
+static bool power_limit_check(void);
+static void compute(struct RegenBraking *regenAttr);
+
+struct RegenBraking regenAttributes;
+static float MAXREGEN = -50.0;
+static float wheelSpeedThreshold = 5.0;
+
+void App_Run_Regen(void) {
+    if (App_Regen_Safety(&regenAttributes)) {
+
+        App_Regen_Activate(regenAttributes.left_inverter_torque, regenAttributes.right_inverter_torque);
+
+    } else {
+
+        App_Regen_Deactivate(regenAttributes.left_inverter_torque, regenAttributes.right_inverter_torque);
+
+    }
+}
+
+bool App_Regen_Safety(struct RegenBraking *regenAttr)
+{
+    if (App_CanRx_BMS_CellTemperatures_MaxCellTemperature_Get() < 45
+    && wheel_speed_in_range() 
+    && power_limit_check()) {
+        compute(regenAttr);
+    }
+
+    regenAttr->left_inverter_torque = 0.0;
+    regenAttr->right_inverter_torque = 0.0;
+
+    return true;
+}
```

Comment:
I don't think these needs to be a new function, since it always returns true

Reply:
I changed this